executor/ddl.go

```diff
@@ -38,12 +39,32 @@ type DDLExec struct {
 	done bool
 }
 
+// toErr converts the error to the ErrInfoSchemaChanged when the schema is outdated.
+func (e *DDLExec) toErr(err error) error {
+	if e.ctx.GetSessionVars().StmtCtx.IsDDLJobReady {
+		return err
+	}
+
+	// Before the DDL job is ready, it encouters an error that may be due to the outdated schema information.
+	// After the DDL job is ready, the ErrInfoSchemaChanged error won't happen because we are getting the schema directly from storage.
+	// So we needn't to consider this condition.
+	// Here we distinguish the ErrInfoSchemaChanged error from other errors.
+	dom := domain.GetDomain(e.ctx)
+	checker := schemachecker.NewSchemaChecker(dom, e.is.SchemaMetaVersion(), nil)
+	schemaInfoErr := checker.Check(e.ctx.Txn().StartTS())
```

Comment:
The check result won't be `ErrInfoSchemaChanged`, because tableIDs is nil.

Reply:
If "ResultFail" is true, the result is ErrInfoSchemaChanged.